actionview/lib/action_view/base.rb

```diff
@@ -175,6 +175,15 @@ def cache_template_loading=(value)
         ActionView::Resolver.caching = value
       end
 
+      def eager_load_templates
+        ActionView::TemplateEagerLoader.eager_load_templates?
+      end
+
+      def eager_load_templates=(value)
+        raise "To eager load view templates config.eager_load should be true" unless !value || Rails.application.config.eager_load
```

Comment:
This is coupling this framework with the application. In one cause where you are using action view outside Rails this will fail. We should move this check up to the Action View Railtie.